iocore/net/BIO_fastopen.cc

```diff
@@ -27,6 +27,67 @@
 
 #include "BIO_fastopen.h"
 
+// For BoringSSL, which for some reason doesn't have this function.
+// (In BoringSSL, sock_read() and sock_write() use the internal
+// bio_fd_non_fatal_error() instead.) #1437
+//
+// The following is copied from
+// https://github.com/openssl/openssl/blob/3befffa39dbaf2688d823fcf2bdfc07d2487be48/crypto/bio/bss_sock.c
```

Comment:
How about just add a function like this? We don't support WINDOWS and don't need those "#ifdef". Just checking error codes looks enough.
```
static int
non_fetal_error(int err)
{
  if (err == EWOULDBLOCK || err == ENOTCONN || err == EINTR || err == EAGAIN || err == EPROTO || err == EINPROGRESS || err == EALREADY) {
    return 1;
  }
  return 0;
}
```